userspace/libsinsp/socket_handler.h

```diff
@@ -1561,22 +1556,33 @@ class socket_data_handler
 			{
 				*(parser_data->m_msg_completed) = true;
 			}
-			else { throw sinsp_exception("Socket handler: parser msg complete null."); }
+			else { throw sinsp_exception("Socket handler (http_msg_completed_callback): parser data m_msg_completed is null."); }
+			if(parser_data->m_http_response)
+			{
+				*(parser_data->m_http_response) = parser->status_code;
+			}
+			else { throw sinsp_exception("Socket handler (http_msg_completed_callback): parser data m_http_response is null."); }
 		}
-		else { throw sinsp_exception("Socket handler: parser or data null."); }
+		else { throw sinsp_exception("Socket handler (http_msg_completed_callback): parser or data null."); }
 		return 0;
 	}
 
 	void init_http_parser()
 	{
+		m_msg_completed = false;
+		m_http_response = -1;
 		http_parser_settings_init(&m_http_parser_settings);
 		m_http_parser_settings.on_body = http_body_callback;
 		m_http_parser_settings.on_message_complete = http_msg_completed_callback;
-		m_http_parser = (http_parser *)std::malloc(sizeof(http_parser));
+		if(!m_http_parser)
+		{
+			m_http_parser = (http_parser *)std::malloc(sizeof(http_parser));
+		}
 		if(m_http_parser)
```

Comment:
Well, I try to behave politely, even when the Titanic is sinking :) But yes, you're right, if malloc fails, not much left to do ...